xarray/tests/test_computation.py

```diff
@@ -1898,3 +1898,86 @@ def test_polyval(use_dask, use_datetime):
     da_pv = xr.polyval(da.x, coeffs)
 
     xr.testing.assert_allclose(da, da_pv.T)
+
+
+@pytest.mark.parametrize(
+    "a, b, ae, be, dim, axis",
+    [
+        [
+            xr.DataArray(np.array([1, 2, 3])),
+            xr.DataArray(np.array([4, 5, 6])),
+            np.array([1, 2, 3]),
+            np.array([4, 5, 6]),
+            None,
+            -1,
+        ],
+        [
+            xr.DataArray(np.array([1, 2])),
+            xr.DataArray(np.array([4, 5, 6])),
+            np.array([1, 2]),
+            np.array([4, 5, 6]),
+            None,
+            -1,
+        ],
+        [  # Test dim in the middle:
+            xr.DataArray(
```

Comment:
This crashes now because xarray returns the cross product with `dims=(time, var, cartesian)`, but numpy returns in the original format `dims=(time, cartesian, var)`. 
It's probably because I do `output_core_dims=[["cartesian"]]` in `apply_ufunc`, but it seems necessary when I `input_core_dims`?
Any tricks to solve this?

Reply:
`xr.dot` had already thought of this, use `transpose` after `apply_ufunc`!